lib/collection/src/collection_manager/optimizers/indexing_optimizer.rs

```diff
@@ -182,27 +186,36 @@ impl IndexingOptimizer {
                     }
                 }
 
-                require_optimization.then_some((*idx, max_vector_size))
+                require_optimization.then_some((*idx, max_vector_size_bytes))
             })
             .collect();
 
         // Select the largest unindexed segment, return if none
         let selected_segment = candidates
             .iter()
-            .max_by_key(|(_, vector_size)| *vector_size);
+            .max_by_key(|(_, vector_size_bytes)| *vector_size_bytes);
         if selected_segment.is_none() {
             return vec![];
         }
         let (selected_segment_id, selected_segment_size) = *selected_segment.unwrap();
 
+        let number_of_segments = segments_read_guard.len();
+
+        // If the number of segments if equal or bigger than the default_segments_number
+        // We want to make sure that we at least do not increase number of segments after optimization
+
+        if number_of_segments < self.default_segments_number {
+            return vec![selected_segment_id];
+        }
```

Comment:
I wonder how well this works if we have multiple indexation jobs (such as 5) at the same time. This is easily possible now with CPU saturation on big CPUs.

Reply:
If we spawn a bunch of those jobs below or above the limit, we might get a disbalance in number of segments again. But I'm afraid we just have to accept that.

The alternative would be to calculate how the number of segments will change considering all ongoing optimizations, so we can make a better estimated guess here. But that's probably more complex than helpful.